### PR TITLE
Add new resource for Service Project for Apphub

### PR DESCRIPTION
**Pull request by eshagoel06**
FIXES https://github.com/hashicorp/terraform-provider-google/issues/17405

```release-note:new-resource
google_apphub_service_project_attachment
```

### CODE REVIEW

File: mmv1/products/apphub/ServiceProject.yaml [hunk 1]
```diff
@@ -1,10 +1,24 @@
+# Copyright 2023 Google Inc.
```
Comment (c2thorn) [inline mmv1/products/apphub/ServiceProject.yaml:1]:
```suggestion
# Copyright 2024 Google Inc.
```

File: mmv1/products/apphub/product.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2023 Google Inc.
```
Comment (c2thorn) [inline mmv1/products/apphub/product.yaml:1]:
```suggestion
# Copyright 2024 Google Inc.
```

Comment (eshagoel06) [reply mmv1/products/apphub/product.yaml]: Done.

File: mmv1/products/apphub/ServiceProject.yaml (hunk 1 above)
Comment (eshagoel06) [reply mmv1/products/apphub/ServiceProject.yaml]: Done. Moved to new file ServiceProjectAttachment.yaml to reflect updated requirements.

**Comment by slevenick** [comment]
/gcbrun

Enabled the API

File: .ci/infra/terraform/main.tf [hunk 3]
```diff
@@ -168,6 +168,7 @@ module "project-services" {
     "apikeys.googleapis.com",
     "appengine.googleapis.com",
     "appengineflex.googleapis.com",
+    "apphub.googleapis.com",
```
Comment (c2thorn) [inline .ci/infra/terraform/main.tf:171]: please remove this like in [`aa5bcff` (#10079)](https://github.com/GoogleCloudPlatform/magic-modules/pull/10079/commits/aa5bcffc499464b874800b275c579c1817482478), it is stopping the build

**Review by c2thorn — CHANGES_REQUESTED** [review]
are you able to see the test failures? looks like we are getting 
```
Error 403: Permission 'apphub.serviceProjectAttachments.attach' denied on resource '//cloudresourcemanager.googleapis.com/projects/tf-test-project-<suffix>' (or it may not exist).
```
@eshagoel06

File: .ci/infra/terraform/main.tf (hunk 3 above)
Comment (eshagoel06) [reply .ci/infra/terraform/main.tf]: Removed.

**Comment by c2thorn** [comment]
> are you able to see the test failures? looks like we are getting
> 
> ```
> Error 403: Permission 'apphub.serviceProjectAttachments.attach' denied on resource '//cloudresourcemanager.googleapis.com/projects/tf-test-project-<suffix>' (or it may not exist).
> ```
> 
> @eshagoel06

@eshagoel06

**Comment by eshagoel06** [comment]
The IAM permissions have been added at org level manually as per https://github.com/GoogleCloudPlatform/magic-modules/pull/10152#pullrequestreview-1925782261

**Comment by eshagoel06** [comment]
> > are you able to see the test failures? looks like we are getting
> > ```
> > Error 403: Permission 'apphub.serviceProjectAttachments.attach' denied on resource '//cloudresourcemanager.googleapis.com/projects/tf-test-project-<suffix>' (or it may not exist).
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > @eshagoel06
> 
> @eshagoel06

Fixed

**Comment by eshagoel06** [comment]
> are you able to see the test failures? looks like we are getting
> 
> ```
> Error 403: Permission 'apphub.serviceProjectAttachments.attach' denied on resource '//cloudresourcemanager.googleapis.com/projects/tf-test-project-<suffix>' (or it may not exist).
> ```
> 
> @eshagoel06

Fixed

**Review by c2thorn — APPROVED** [review]
LGTM considering all you can really do with this resource is create it given URL parameters.

**Pull request merged**